tensorboard/plugins/debugger/debugger_server_test.py

```diff
@@ -0,0 +1,260 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests the debugger data server, which receives and writes debugger events."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import json
+
+import mock
+import numpy
+from six.moves import xrange  # pylint: disable=redefined-builtin
+import tensorflow as tf
+
+from google.protobuf import json_format
+from tensorflow.core.debug import debugger_event_metadata_pb2
+
+from tensorboard.plugins.debugger import constants
+from tensorboard.plugins.debugger import debugger_server_lib
+from tensorboard.plugins.debugger import numerics_alert
+
+
+class FakeEventsWriterManager(object):
+  """An events writer manager that tracks events that would be written.
+
+  During normal usage, the debugger data server would write events to disk.
+  Unfortunately, this test cannot depend on TensorFlow's record reader due to
+  GRPC library conflicts (b/35006065). Hence, we use a fake EventsWriter that
+  keeps track of events that would be written to disk.
+  """
+
+  def __init__(self, events_output_list):
+    """Constructs a fake events writer, which appends events to a list.
+
+    Args:
+      events_output_list: The list to append events that would be written to
+        disk.
+    """
+    self.events_written = events_output_list
+
+  def dispose(self):
+    """Does nothing. This implementation creates no file."""
```

Comment:
I know this is technically valid, but can you add a `pass` statement here, or will pylint complain?